src/binding/python/openpmd_api/DataFrame.py

```diff
@@ -0,0 +1,65 @@
+"""
+This file is part of the openPMD-api.
+
+Copyright 2021 openPMD contributors
+Authors: Axel Huebl
+License: LGPLv3+
+"""
+import numpy as np
+try:
+    import pandas as pd
+    found_pandas = True
+except ImportError:
+    print("pandas NOT found. Install pandas for DataFrame support.")
+    found_pandas = False
+
+
+def particles_to_dataframe(particle_species, slice=None):
```

Comment:
Potential extra arguments:
- `records = iterable`: a list of strings or regex expressions to load only specific records of the species
- maybe an inverse parameter to deselect specific records from loading